tensorbay/client/requests.py

```diff
@@ -332,7 +332,13 @@ def multithread_upload(
         if callback is not None:
             multi_callback = MultiCallbackTask(function=function, callback=callback)
             function = multi_callback.work
-        futures = [executor.submit(function, argument) for argument in arguments]
+
+        futures: List[Any] = []
+        for argument in arguments:
+            if isinstance(argument, (tuple, list)):
+                futures.append(executor.submit(function, *argument))
```

Comment:
the argument is unpacked in the input lambda (dataset.py L571), why unpack again here?